galsim_extra/image_lattice.py

```diff
@@ -0,0 +1,224 @@
+        self.border_ratio = params.get("border_ratio", 1)
+
+        size = params.get('size',0)
+        full_xsize = params.get('xsize',size)
+        full_ysize = params.get('ysize',size)
+
+        if (full_xsize <= 0) or (full_ysize <= 0):
+            raise GalSimConfigError(
+                "Both image.xsize and image.ysize need to be defined and > 0.")
+
+        # If image_force_xsize and image_force_ysize were set in config, make sure it matches.
+        if ( ('image_force_xsize' in base and full_xsize != base['image_force_xsize']) or
+             ('image_force_ysize' in base and full_ysize != base['image_force_ysize']) ):
+            raise GalSimConfigError(
+                "Unable to reconcile required image xsize and ysize with provided "
+                "xsize=%d, ysize=%d, "%(full_xsize,full_ysize))
+
+        return full_xsize, full_ysize
+
+
+    def buildImage(self, config, base, image_num, obj_num, logger):
+        """
+        Build an Image consisting of a tiled array of postage stamps.
+
+        Parameters:
+            config:     The configuration dict for the image field.
+            base:       The base configuration dict.
+            image_num:  The current image number.
+            obj_num:    The first object number in the image.
+            logger:     If given, a logger object to log progress.
+
+        Returns:
+            the final image and the current noise variance in the image as a tuple
+        """
+        full_xsize = base['image_xsize']
+        full_ysize = base['image_ysize']
+        wcs = base['wcs']
+
+        dtype = _ParseDType(config, base)
+        full_image = Image(full_xsize, full_ysize, dtype=dtype)
+        full_image.setOrigin(base['image_origin'])
+        full_image.wcs = wcs
+        full_image.setZero()
+        base['current_image'] = full_image
+
+        # Make a list of ix,iy values according to the specified order:
+        rng = np.random.default_rng()  # TODO: how to best handle RNGs with galsim?
+        # TODO: pull lattice vectors from config
+        v1 = np.asarray([1, 0], dtype=float)
+        v2 = np.asarray([np.cos(np.radians(120)), np.sin(np.radians(120))], dtype=float)
+        x_lattice, y_lattice = build_lattice(full_xsize-1, full_ysize-1, self.sep, base["pixel_scale"], v1, v2, self.rotation, self.border_ratio)
+        nobjects = len(x_lattice)
+
+        # Define a 'image_pos' field so the stamps can set their position appropriately in case
+        # we need it for PowerSpectum or NFWHalo.
+        config['image_pos'] = {
+            'type' : 'XY',
+            'x' : {
+                'type' : 'List',
+                'items' : x_lattice
+            },
+            'y' : {
+                'type' : 'List',
+                'items' : y_lattice
+            }
+        }
+
+        stamps, current_vars = BuildStamps(
+            nobjects, base, logger=logger, obj_num=obj_num,do_noise=False
+        )
+
+        base['index_key'] = 'image_num'
```

Comment:
Can you comment on what this line does and why it is here?

Reply:
I think he's just following what the Tiled builder does.  This makes it so any sequencing or random values in this part of the code use the image_num for sequencing and the image rng for random values.

Reply:
So this will override anything previously set?

Reply:
Yes.  It was set to obj_num during buildStamps.  Now that we're back in the image builder, we want it to be set back to image_num.

Reply:
As Mike mentioned, I was following the Tiled builder. It could definitely be the case that we don't want to do this

Reply:
Yeah not saying it is wrong. Asking how it works.